net/developer-guide/working-with-outlook-storage-files/working-with-messages-in-a-pst-file/_index.md

```diff
@@ -127,18 +621,83 @@ Sometimes its required to mark items that are processed with in the PST file. As
+		MapiProperty namedProperty2 = new MapiNamedProperty(GenerateNamedPropertyTag(1, MapiPropertyType.PT_LONG),0x0000870C,new Guid("0006200A-0000-0000-C000-000000000046"),BitConverter.GetBytes(0));
+		newProperties.Add(namedProperty1.Tag, namedProperty1);
+		newProperties.Add(namedProperty2.Tag, namedProperty2);
+		newProperties.Add(property.Tag, property);
+		testFolder.ChangeMessages(testFolder.EnumerateMessagesEntryId(), newProperties);
+	}
+}
+
+private static long GenerateNamedPropertyTag(long index, MapiPropertyType dataType)
+{
+	return (((0x8000 | index) << 16) | (long)dataType) & 0x00000000FFFFFFFF;
+}
+```
 
-
-{{< gist "aspose-email" "9e8fbeb51a8cbc4129dc71ca8cd55f0b" "Examples-CSharp-PST-UpdatePSTCustomProperites-UpdatePSTCustomProperites.cs" >}}
 ## **Extract Attachments without Extracting Complete Message**
 Aspose.Email API can be used to extract attachments from PST messages without extracting the complete message first. The ExtractAttachments method of IEWSClient can be used to do this. The following code snippet shows you how to extract attachments without extracting complete message.
 
+```csharp
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-.NET
+string dataDir = RunExamples.GetDataDir_Outlook();
+
+using (PersonalStorage personalstorage = PersonalStorage.FromFile(dataDir + "Outlook.pst"))
+{
+    FolderInfo folder = personalstorage.RootFolder.GetSubFolder("Inbox");
+
+    foreach (var messageInfo in folder.EnumerateMessagesEntryId())
+    {
+        MapiAttachmentCollection attachments = personalstorage.ExtractAttachments(messageInfo);
+
+        if (attachments.Count != 0)
+        {
+            foreach (var attachment in attachments)
+            {
+                if (!string.IsNullOrEmpty(attachment.LongFileName))
+                {
+                    if (attachment.LongFileName.Contains(".msg"))
+                    {
+                        continue;
+                    }
+                    else
+                    {
+                        attachment.Save(dataDir + @"\Attachments\" + attachment.LongFileName);
+                    }
+                }
+            }
+        }
+    }
+}
+```
 
-
-{{< gist "aspose-email" "9e8fbeb51a8cbc4129dc71ca8cd55f0b" "Examples-CSharp-Outlook-ExtractAttachmentsFromPSTMessages-ExtractAttachmentsFromPSTMessages.cs" >}}
 ## **Adding Files to PST**
 Microsoft Outlook's key functionality is managing emails, calendars, tasks, contacts and journal entries. In addition, files can also be added to a PST folder and the resulting PST keeps record of the documents added. Aspose.Email provides the facility to add files to a folder in the same way in addition to adding messages, contacts, tasks and journal entries to PST. The following code snippet shows you how to add documents to a PST folder using Aspose.Email.
 
+```csharp
+/ For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-.NET

```

Comment:
`// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-.NET`